src/Maestro/Maestro.Data/BuildAssetRegistryContext.cs

```diff
@@ -327,6 +327,29 @@ where buildIds.Contains(build.Id)
                 dict[edge.BuildId].DependentBuildIds.Add(edge);
             }
 
+            // Gather subscriptions used by this build.
+            Build primaryBuild = Builds.First(b => b.Id == buildId);
+            var validSubscriptions = await Subscriptions.Where(s => (s.TargetRepository == primaryBuild.AzureDevOpsRepository || s.TargetRepository == primaryBuild.GitHubRepository) &&
+                                                                    (s.TargetBranch == primaryBuild.AzureDevOpsBranch || s.TargetBranch == primaryBuild.GitHubBranch ||
+                                                                     $"refs/heads/{s.TargetBranch}" == primaryBuild.AzureDevOpsBranch || $"refs/heads/{s.TargetBranch}" == primaryBuild.GitHubBranch)).ToListAsync();
+
+            // Use the subscriptions to determine what channels are relevant for this build, so just grab the unique channel ID's from valid suscriptions
+            var channelIds = validSubscriptions.GroupBy(x => x.ChannelId).Select(y => y.First()).Select(s => s.ChannelId);
+
+            // Acquire list of builds in valid channels
+            var channelBuildIds = await BuildChannels.Where(b => channelIds.Any(c => c == b.ChannelId)).Select(s => s.BuildId).ToListAsync();
+            var possibleBuilds = await Builds.Where(b => channelBuildIds.Any(c => c == b.Id)).ToListAsync();
+
+            // Calculate total number of builds that are newer.
+            foreach (var id in dict.Keys)
```

Comment:
This can just be `foreach (var (id, build) in dict)` to skip the keys enumeration and indexer.

Reply:
This is neat, but apparently arcade-services isn't using a new enough version of C# and this just causes build errors.